### DISCUSSION TITLE
Getting error code 401 for my API key, but key works everywhere else?

### DISCUSSION THREAD

**Discussion by fabiendostie** [danielmiessler/Fabric#145] [Q&A] [answered]
When I first installed fabric i entered a different API key but it wasnt working because I didn't have any credits. so I bought some and generated a new api key. I came back to my terminal and did ```fabric --setup``` again it asked for my openai api key, which i provided:
```Welcome to Fabric. Let's get started.
Please enter your OpenAI API key
sk-xkM------------------------------ 
Updating patterns...
Downloaded zip file successfully.
Extracted zip file successfully.
Patterns updated successfully.
Aliases added successfully. Please restart your terminal to use them.```

Now I get this error whenever I try to use fabric:

```
Error: Error code: 401 - {'error': {'message': 'Incorrect API key provided: sk-9X7xF***************************************dOLB. You can find your API key at https://platform.openai.com/account/api-keys.', 'type': 'invalid_request_error', 'param': None, 'code': 'invalid_api_key'}}
```

My issue is that the new key I provided doesn't seem to get registered by fabric because my new key does NOT start with : **sk-9X7xF it starts with sk-xkM?** 
Anyone has any suggestions for me?

I have tested my api key with python:

```
import openai

def check_openai_api_key(api_key):
    client = openai.OpenAI(api_key=api_key)
    try:
        client.models.list()
    except openai.AuthenticationError:
        return False
    else:
        return True

OPENAI_API_KEY = "sk-xkMr******************************0uFN"

if check_openai_api_key(OPENAI_API_KEY):
    print("Valid OpenAI API key.")
else:
    print("Invalid OpenAI API key.")
```


I get:
```
examples git:(master) ✗ /usr/local/bin/python3 /Users/lefab/Documents/scripts/OpenAi_keytest/open_ai_apikey_test.py
Valid OpenAI API key.
```

Thanks for the help

**Comment by xssdoctor** [comment, marked as answer]
You can manually change the key. It’s in ~/.config/fabric/.env

**Comment by fabiendostie** [comment]
I've literally just found out about this!
It now works! I'll change my post to resolved!

Thanks for your help

--
. | Fabien Dostie |
.///Coloriste Resolve | Montage Online VR-HD
.//Compositing | VR Stitching + editing
./DIT | DMT | V/A AQTIS 514 IATSE 667
.Tech. Scène IATSE 56
C: 438.357.3462 W: fabiendostie.com

**Comment by sanebg** [comment]
This is stupid, the app thinks I have entered API key I never heard of. I double checked all the folders for the settings. This is my 8th time for the past 6 months trying to run this abomination. The most user-unfriendly app ever.
